Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -1215,9 +1215,11 @@ instance finiteDimensional_iSup_of_finite {ι : Type _} {t : ι → Intermediate
 #align intermediate_field.finite_dimensional_supr_of_finite IntermediateField.finiteDimensional_iSup_of_finite
 
 instance finiteDimensional_iSup_of_finset {ι : Type _} {f : ι → IntermediateField K L}
-    {s : Finset ι} [h : ∀ i ∈ s, FiniteDimensional K (f i)] :
+    /-Porting note: changed `h` from `∀ i ∈ s, FiniteDimensional K (f i)` because this caused an
+      error.-/
+    {s : Finset ι} [h : ∀ i, FiniteDimensional K (f i)] :
     FiniteDimensional K (⨆ i ∈ s, f i : IntermediateField K L) := by
```

Comment:
This seems strictly less powerful; should probably check what happened here before landing

Reply:
```lean
instance finiteDimensional_iSup_of_finset {ι : Type _} {f : ι → IntermediateField K L}
    /-Porting note: changed `h` from `∀ i ∈ s, FiniteDimensional K (f i)` because this caused an
      error.-/
    {s : Finset ι} (h : ∀ i, i ∈ s → FiniteDimensional K (f i)) :
    FiniteDimensional K (⨆ i ∈ s, f i : IntermediateField K L) := by
  haveI : ∀ i : { i // i ∈ s }, FiniteDimensional K (f i) := fun i => h i i.2
  have : (⨆ i ∈ s, f i) = ⨆ i : { i // i ∈ s }, f i :=
    le_antisymm (iSup_le fun i => iSup_le fun h => le_iSup (fun i : { i // i ∈ s } => f i) ⟨i, h⟩)
      (iSup_le fun i => le_iSup_of_le i (le_iSup_of_le i.2 le_rfl))
  exact this.symm ▸ IntermediateField.finiteDimensional_iSup_of_finite
```
works and its mathematically equivalent then the mathlib3 version. Using `[h : ...]` causes the error
```
invalid parametric local instance, parameter with type
  i ∈ s
does not have forward dependencies, type class resolution cannot use this kind of local instance because it will not be able to infer a value for this parameter.
```

Reply:
But now my instance is not found in two lemmas below. We can maybe have both, yours as instance and mine as a theorem in case someone really needs it.